diff --git a/llvm/include/llvm/LTO/Config.h b/llvm/include/llvm/LTO/Config.h
index 57bba5e..9ac5c73 100644
--- a/llvm/include/llvm/LTO/Config.h
+++ b/llvm/include/llvm/LTO/Config.h
@@ -93,7 +93,7 @@ struct Config {
   /// Statistics output file path.
   std::string StatsFile;
 
-  bool ShouldDiscardValueNames = true;
+  bool ShouldDiscardValueNames = false;
   DiagnosticHandlerFunction DiagHandler;
 
   /// If this field is set, LTO will write input file paths and symbol
